hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/PathData.java

```diff
@@ -308,6 +308,30 @@ private String getStringForChildPath(Path childPath) {
   protected enum PathType { HAS_SCHEME, SCHEMELESS_ABSOLUTE, RELATIVE };
   
   /**
+   * Resolves an actual file system for the given {@link PathData}
+   * through any symlinks or mount points.
+   * @param src {@link PathData} to be resolved
+   * @param conf the hadoop configuration
+   * @return new {@link PathData} if a fs changed, or the same object otherwise
+   * @throws FileNotFoundException when the path does not exist
+   * @throws IOException from #{@link FileSystem} implementations
+   */
+  public static PathData resolveFsPath(PathData src, Configuration conf)
+      throws IOException {
+    Path resolvedPath = src.fs.resolvePath(src.path);
+    if (resolvedPath.equals(src.path)) {
+      return src;
+    }
+    FileSystem resolvedFs = resolvedPath.getFileSystem(conf);
+    if (resolvedFs.equals(src.fs)) {
+      return src;
+    }
+    // FileStatus contains unresolved path thus should be replaced too
+    FileStatus resolvedStat = resolvedFs.getFileStatus(resolvedPath);
+    return new PathData(resolvedFs, resolvedPath.toString(), resolvedStat);
```

Comment:
Perhaps you can use return new PathData(resolvedPath.toString(), conf); to simplify the code

Reply:
or return new PathData(resolvedFs, resolvedPath.toString());